### PR TITLE
proto level validation

### PR DESCRIPTION
**Pull request by gertd**
This PR adds fields level validation constraints to the protobuf contracts

### CODE REVIEW

File: proto/aserto/directory/common/v3/common.proto [hunk 1]
```diff
@@ -7,84 +7,309 @@ option go_package = "github.com/aserto-dev/go-directory/aserto/directory/common/
...
+    ];
+    // external object identifier (cs-string, no spaces or tabs)
+    string id                                           = 2
+    [
+        (buf.validate.field) = {
+            required: true,
+            cel: {
+                id: "object.id"
+                message: "cannot contain any spaces or other whitespace characters"
+                expression: "this.matches('^[^\\t\\n\\f\\r ]+$')"
```
Comment (ronenh) [inline proto/aserto/directory/common/v3/common.proto:38]: You should be able to use `\S` to match non-whitespace characters:
`this.matches('^\\S+$')`

Comment (gertd) [reply proto/aserto/directory/common/v3/common.proto:38]: I know, that triggers a bug in protovalidate.

Comment (gertd) [reply proto/aserto/directory/common/v3/common.proto:38]: @ronenh see https://bufbuild.slack.com/archives/CRZ680FUH/p1690577643575079 and https://github.com/gertd/grpc-test

Comment (gertd) [reply proto/aserto/directory/common/v3/common.proto]: Double escaping provided the workaround, updated the PR accordingly.
Still unclear why `'^[\\S]+$'` fails and `'^[^\\t\\n\\f\\r ]+$'` succeeds

**Pull request merged**